core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -237,6 +237,12 @@ private[spark] class AkkaRpcEnv private[akka] (
   }
 
   override def toString: String = s"${getClass.getSimpleName}($actorSystem)"
+
+  def deserialize[T](deserializationAction: () => T): T = {
+    JavaSerializer.currentSystem.withValue(actorSystem.asInstanceOf[ExtendedActorSystem]) {
```

Comment:
I see, because now we no longer pass akka's `Serialization`, which has information about the actor system, into `PersistenceEngine`, so here we ensure that we're using the actor system's serializer.

But more generally, since we always serialize with `JavaSerializer` in the new code, why can't we always deserialize with the same thing? I just find it a little strange that we have to pass a closure into this method.